src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2504,6 +2532,14 @@ theorem getLsbD_replicate {n w : Nat} (x : BitVec w) :
       simp only [show ¬i < w * n by omega, decide_False, cond_false, hi, Bool.false_and]
       apply BitVec.getLsbD_ge (x := x) (i := i - w * n) (ge := by omega)
 
+@[simp]
+theorem getElem_replicate {n w : Nat} (x : BitVec w) (h : i < w * n) :
+    (x.replicate n)[i] = x.getLsbD (i % w) := by
```

Comment:
This could be `= x[i]'(proof goes here)`.

Reply:
I had to introduce an `if w = 0 then false else ...` but otherwise this change is a great improvement.